build_tools/cmake/iree_check_test.cmake

```diff
@@ -0,0 +1,153 @@
+  endif()
+
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  set(_MODULE_TARGET_NAME "${_RULE_NAME}_module")
+
+  iree_bytecode_module(
+    NAME
+      "${_MODULE_TARGET_NAME}"
+    SRC
+      "${_RULE_SRC}"
+    FLAGS
+      "-iree-mlir-to-vm-bytecode-module"
+      "--iree-hal-target-backends=${_RULE_TARGET_BACKEND}"
+    TESTONLY
+  )
+
+  # TODO(bf/146898896): Do this in iree_bytecode_module and avoid having to
+  # reach into the internals.
+  set(_GENERATED_MODULE_NAME "${_MODULE_TARGET_NAME}.module")
+  add_custom_target(
+    "${_NAME}" ALL
+     DEPENDS
+       "${_GENERATED_MODULE_NAME}"
+  )
+
+  string(REPLACE "_" "/" _PACKAGE_PATH ${_PACKAGE_NAME})                                                                                                    
+  set(_NAME_PATH "${_PACKAGE_PATH}:${_RULE_NAME}") 
+
+  add_test(
+    NAME
+      "${_NAME_PATH}"
+    COMMAND
+      "${CMAKE_SOURCE_DIR}/build_tools/cmake/run_test.${IREE_HOST_SCRIPT_EXT}"
+      "$<TARGET_FILE:iree_modules_check_iree-check-module>"
+      "--input_file=${CMAKE_CURRENT_BINARY_DIR}/${_GENERATED_MODULE_NAME}"
+      "--driver=${_RULE_DRIVER}"
+      "${_RULE_ARGS}"
+  )
+
+  set_property(TEST "${_NAME_PATH}" PROPERTY REQUIRED_FILES "${_GENERATED_MODULE_NAME}")
+  set_property(TEST "${_NAME_PATH}" PROPERTY ENVIRONMENT "TEST_TMPDIR=${_NAME}_test_tmpdir")
+  set_property(TEST "${_NAME_PATH}" PROPERTY LABELS "${_PACKAGE_PATH}")
+endfunction()
+
+
+# iree_check_test_suite()
+#
+# Creates a test suite of iree-check-module tests.
+#
+# Mirrors the bzl rule of the same name.
+#
+# One test is generated per source and backend/driver pair.
+# Parameters:
+#   NAME: name of the generated test suite.
+#   SRCS: source mlir files containing the module.
+#   TARGET_BACKENDS: backends to compile the module for. These form pairs with
+#       the DRIVERS argument (due to cmake limitations they are separate list
+#       arguments). The lengths must exactly match. If no backends or drivers are
+#       specified, a test will be generated for every supported pair.
+#   DRIVERS: drivers to run the module with. These form pairs with the
+#       TARGET_BACKENDS argument (due to cmake limitations they are separate list
+#       arguments). The lengths must exactly match. If no backends or drivers are
+#       specified, a test will be generated for every supported pair.
+#   ARGS: additional args to pass to the underlying iree-check-module tests. The
+#       driver and input file are passed automatically. To use different args per
+#       test, create a separate suite or iree_check_test.
+function(iree_check_test_suite)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME"
+    "SRCS;TARGET_BACKENDS;DRIVERS;ARGS"
+    ${ARGN}
+  )
+  if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  if(NOT DEFINED _RULE_TARGET_BACKENDS AND NOT DEFINED _RULE_DRIVERS)
+    set(_RULE_TARGET_BACKENDS "vmla" "vulkan-spirv")
+    set(_RULE_DRIVERS "vmla" "vulkan")
```

Comment:
The parameter description only mentions "every supported pair" as default. So what about llvmjit?

Reply:
This matches the bazel one right now. I had llvmjit in there, but it got deleted in the dynamic shapes feature branch merge :-/ I think it might be fair to call that unsupported right now? I need to follow up on why it was removed.

Reply:
> This matches the bazel one right now.

That's all the excuse you need ;)

Reply:
:-D